matrix_client/admin_api.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+from matrix_client.api import MatrixHttpApi
+try:
+    basestring
+except NameError:
+    basestring = str
+
+
+class MatrixHttpAdminApi(MatrixHttpApi):
+    """Extends Matrix API with admin calls.
+
+    Examples:
+        Create a client and send a message::
+
+            matrix = MatrixHttpAdminApi("https://matrix.org", token="foobar")
+            response = admin_api.shutdown_room(
+                "!DgvjtOljKujDBrxyHk:matrix.org",
+                "@admin:matrix.org",
+                room_name="New room",
+                message="Old room closed by admin"
+            )
+    """
+    def purge_history(self, room_id, event_id):
+        """Perform /admin/purge_hostory.
+        Admin api part.
+        Args:
+            room_id (str): Room_id to purge.
+            event_id (str or int): Event_id or ts to purge before.
+        """
+        if isinstance(event_id, basestring):
+            content = {
+                "delete_local_events": True,
+                "purge_up_to_event_id": event_id
+            }
+        else:
+            content = {
+                "delete_local_events": True,
+                "purge_up_to_ts": int(event_id)
+            }
+        return self._send("POST", "/admin/purge_history/%s" % room_id, content)
+
+    def purge_history_status(self, purge_id):
+        """Perform /admin/purge_history_status.
+        Admin api part.
```

Comment:
I doubt this line is useful, but if you want to keep it it needs a newline after (applies to every docstrings).